src/dojos/cd-find-dojo.vue

```diff
@@ -1,24 +1,39 @@
-        <h1 class="cd-find-dojo__panel-form-header">{{ $t('Find a Dojo to attend') }}</h1>
-        <p class="cd-find-dojo__panel-form-info">{{ $t('Learn technology in an informal, creative and social environment. Find a dojo near you.') }}</p>
+    <div v-bind:class="searchExecuted ? 'cd-find-dojo__panel--reduced' : 'cd-find-dojo__panel'">
+      <form v-bind:class="searchExecuted ? 'cd-find-dojo__panel-form--reduced' : 'cd-find-dojo__panel-form'" @submit.prevent="$router.push({ query: { q: searchCriteria } });">
+        <h1 v-if="!searchExecuted" class="cd-find-dojo__panel-form-header">{{ $t('Find a Dojo to attend') }}</h1>
+        <p v-if="!searchExecuted" class="cd-find-dojo__panel-form-info">{{ $t('Learn technology in an informal, creative and social environment. Find a dojo near you.') }}</p>
+        <div v-if="searchExecuted" class="cd-find-dojo__panel-form-header--reduced hidden-xs">{{ $t('{total} Dojos found near {query}', {total: dojos.length, query: queryString}) }}</div>
+        <div v-if="searchExecuted" class="cd-find-dojo__panel-form-header--reduced visible-xs">{{ $t('{total} Dojos found', {total: dojos.length}) }}</div>
         <div class="cd-find-dojo__panel-form-search">
-          <div class="cd-find-dojo__panel-form-search-input">
+          <div v-bind:class="searchExecuted ? 'cd-find-dojo__panel-form-search-input--reduced' : 'cd-find-dojo__panel-form-search-input'">
             <input type="text" name="addressSearch" class="form-control input-lg" :placeholder="$t('Enter your city or locality')" v-model="searchCriteria" autofocus>
           </div>
-          <div class="cd-find-dojo__panel-form-search-submit">
+          <div class="cd-find-dojo__panel-form-search-submit hidden-xs">
             <input type="submit" class="btn btn-lg" :value="$t('Search for Dojos')"/>
           </div>
+          <div v-if="!searchExecuted" class="cd-find-dojo__panel-form-search-submit visible-xs">
+            <input type="submit" class="btn btn-lg" :value="$t('Search for Dojos')"/>
+          </div>
+          <span v-else class="cd-find-dojo__panel-form-search-submit--reduced visible-xs">
+            <button type="submit" class="btn btn-lg fa fa-arrow-right"/>
+          </span>
         </div>
-        <router-link tag="button" :to="{ query: { currentLocation: true } }" class="cd-find-dojo__panel-form-detect-location">
+        <router-link tag="button" :to="{ query: { currentLocation: true } }" class="cd-find-dojo__panel-form-detect-location hidden-xs">
+          <i class="fa fa-location-arrow" aria-hidden="true" v-show="!detectingLocation"></i>
+          <i class="fa fa-spinner fa-spin" aria-hidden="true" v-show="detectingLocation"></i>
+          {{ $t('Detect my location') }}
+        </router-link>
+        <router-link tag="button" :to="{ query: { currentLocation: true } }"
+          v-bind:class="searchExecuted ? 'cd-find-dojo__panel-form-detect-location--reduced' : 'cd-find-dojo__panel-form-detect-location'"
```

Comment:
Same as above, `--reduced` should only make modifications, not provide an entirely different set of styles.